tests/jballerina-unit-test/src/test/resources/test-src/query/on-clause.bal

```diff
@@ -0,0 +1,31 @@
+import ballerina/io;
+
+type Person record {|
+    string firstName;
+    string lastName;
+    int age;
+|};
+
+function testSimpleOnClause() returns Person[] {
+
+    Person p1 = {firstName: "Alex", lastName: "George", age: 23};
+    Person p2 = {firstName: "Ranjan", lastName: "Fonseka", age: 30};
+    Person p3 = {firstName: "John", lastName: "David", age: 33};
+
+    Person[] personList = [p1, p2, p3];
+
+    Person[] outputPersonList =
+            from var person in personList
+            on printHello()
```

Comment:
Better to use a proper example. This shouldn't be allowed and type checker should only allow following [1];

`join-on-condition := "on" expression "equals" expression`

[1] https://github.com/ballerina-platform/ballerina-spec/issues/435